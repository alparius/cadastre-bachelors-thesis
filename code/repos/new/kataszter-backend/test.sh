#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
