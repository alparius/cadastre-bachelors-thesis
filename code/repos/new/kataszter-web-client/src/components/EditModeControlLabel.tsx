import React from "react";
import { Responsive, Segment } from "semantic-ui-react";

import "../static/priestProfile.css";
import EditModeControlLabelButtons from "./EditModeControlLabelButtons";

interface IEditModeControlLabelProps {
    text: string;
    onSave(): any;
    onRevertChanges(): any;
}

const EditModeControlLabel: React.FC<IEditModeControlLabelProps> = (props: IEditModeControlLabelProps) => {
    const { text, onSave, onRevertChanges } = props;

    return (
        <>
            {/* Computer view */}
            <Responsive as={Segment} minWidth={769} tertiary id="editControlLabelComputer">
                <EditModeControlLabelButtons text={text} onSave={onSave} onRevertChanges={onRevertChanges} />
            </Responsive>

            {/* Mobile view */}
            <Responsive as={Segment} maxWidth={768} tertiary id="editControlLabelMobile">
                <EditModeControlLabelButtons text={text} onSave={onSave} onRevertChanges={onRevertChanges} />
            </Responsive>
        </>
    );
};

export default EditModeControlLabel;
