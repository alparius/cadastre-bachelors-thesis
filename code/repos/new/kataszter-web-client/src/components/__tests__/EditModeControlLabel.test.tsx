/*===================
  EditModeControlLabel.unit.test.tsx
 ===================*/
import React from "react";

import { createComponentWithIntlAndRouter, mountWithIntl } from "../../util/test/test-wrapper";
import EditModeControlLabel from "../EditModeControlLabel";

const handlers = {
    onSave: jest.fn(),
    onRevertChanges: jest.fn()
};

describe("EditModeControlLabel element", () => {
    it("should render a EditModeControlLabel container", () => {
        const container = createComponentWithIntlAndRouter(<EditModeControlLabel text="profile.editModeText" {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call the onSave function when clicked on saveChanges tabs", () => {
        const container = mountWithIntl(<EditModeControlLabel text="profile.editModeText" {...handlers} />);
        container.find("#saveChanges").at(0).simulate("click");
        expect(handlers.onSave).toHaveBeenCalledTimes(1);
    });

    it("should call the onRevertChanges function when clicked on revertChanges button", () => {
        const container = mountWithIntl(<EditModeControlLabel text="profile.editModeText" {...handlers} />);
        container.find("#revertChanges").at(0).simulate("click");
        expect(handlers.onRevertChanges).toHaveBeenCalledTimes(1);
    });
});
