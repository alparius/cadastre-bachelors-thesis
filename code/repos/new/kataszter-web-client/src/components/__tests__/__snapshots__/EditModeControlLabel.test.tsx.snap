// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditModeControlLabel element should render a EditModeControlLabel container 1`] = `
<div
  className="ui tertiary segment"
  id="editControlLabelComputer"
>
  <div
    className="ui stackable two column grid"
  >
    <div
      className="middle aligned column"
      id="editModeText"
    >
      You are in edit mode.
    </div>
    <div
      className="column"
    >
      <div
        className="ui two column grid"
      >
        <div
          className="column"
        >
          <button
            className="ui violet fluid button"
            id="saveChanges"
            onClick={[Function]}
          >
            Save Changes
          </button>
        </div>
        <div
          className="column"
        >
          <button
            className="ui grey fluid button"
            id="revertChanges"
            onClick={[Function]}
            value={false}
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
